client/my-sites/email/email-management/home/email-no-domain.tsx

```diff
@@ -1,25 +1,44 @@
 import { isFreePlan } from '@automattic/calypso-products';
-import { localize } from 'i18n-calypso';
-import PropTypes from 'prop-types';
+import { useTranslate } from 'i18n-calypso';
 import { useSelector } from 'react-redux';
 import Illustration from 'calypso/assets/images/domains/domain.svg';
 import EmptyContent from 'calypso/components/empty-content';
-import { recordInboxUpsellTracksEvent } from 'calypso/my-sites/email/email-management/home/utils';
+import TrackComponentView from 'calypso/lib/analytics/track-component-view';
+import { recordEmailUpsellTracksEvent } from 'calypso/my-sites/email/email-management/home/utils';
 import { hasDomainCredit } from 'calypso/state/sites/plans/selectors';
+import { SiteData } from 'calypso/state/ui/selectors/site-data';
+import type { AppState } from 'calypso/types';
 
-const noop = () => {};
+const EmailNoDomain = ( { selectedSite, source }: { selectedSite: SiteData; source: string } ) => {
+	const translate = useTranslate();
 
-const EmailNoDomain = ( { selectedSite, translate, source } ) => {
-	const hasAvailableDomainCredit = useSelector( ( state ) =>
+	const hasAvailableDomainCredit = useSelector( ( state: AppState ) =>
 		hasDomainCredit( state, selectedSite.ID )
 	);
 
 	const isFreePlanProduct = isFreePlan( selectedSite?.plan?.product_slug ?? null );
 
-	const trackEvent =
-		source != null
-			? () => recordInboxUpsellTracksEvent( isFreePlanProduct ? 'plan' : 'domain' )
-			: noop;
+	const trackEvent = () => {
+		recordEmailUpsellTracksEvent( source, isFreePlanProduct ? 'plan' : 'domain' );
+	};
+
+	const trackImpression = ( noDomainContext: string ) => {
+		if ( isFreePlanProduct === null || hasAvailableDomainCredit === null ) {
```

Comment:
`isFreePlanProduct` is never going to be `null`, right?

Digging into `hasDomainCredit` and its descendants, it also seems unlikely that `hasAvailableDomainCredit` would ever be `null`. And even if it was, are we sure that we don't want to track the impression at all then..?

Reply:
Right, so the reason I have those checks is because `trackImpression` was being called multiple times, as the component was being mounted. Maybe there are better ways to address that?

Reply:
`trackImpression` can actually be called multiple times, so long as we only make one network request to Tracks. When I remove this entire if statement and test the code, I still only see one network request to Tracks. This is because the `TrackComponentView` is only being *mounted* once, even if `EmailNoDomain` renders multiple times.

So basically, we should be able to remove this if statement without any trouble!

@AllTerrainDeveloper's [Chrome extension](https://github.com/Automattic/tracks-chrome-extension) is actually helpful when testing this, btw.